tests/functional/client/testClientConnections.js

```diff
@@ -112,27 +113,37 @@ describe('Client connections', async function test() {
             }
             assert.strictEqual(res, 'OK');
 
-            // Requests made with async.times will stay open when they occur
-            // during the failover window so and async.race is used to resolve
-            async.race([
-                () => setTimeout(() => this.loadgen.emit('finished'), 3000),
-                () => async.times(
-                    100,
-                    () => makeRequest(this, done),
-                    () => this.loadgen.emit('finished'),
-                ),
+            return async.race([
+                cb => {
+                    setTimeout(() => {
+                        this.loadgen.emit('finished');
+                        cb();
+                    }, 3000);
+                    return undefined;
+                },
+                cb => {
+                    async.times(
+                        100,
+                        (n, next) => makeRequest(this, next),
+                        err => {
+                            if (err) {return cb(err);}
+                            this.loadgen.emit('finished');
+                            return cb();
+                        },
+                    );
+                    return undefined;
+                },
             ]);
         });
 
-        sentinelSub.on('message', (chan, message) => {
-            // wait until the old master is added as a replica so any stale connections would be transfered
+        // Add return statement here
+        return sentinelSub.once('message', (chan, message) => {
```

Comment:
```suggestion
        sentinelSub.once('message', (chan, message) => {
            // wait until the old master is added as a replica so any stale connections would be transferred
```

Also, what's the reason of using `once` versus `on`? did we face errors causing callbacks called twice? If yes, maybe better to understand why